docs/validation/index.md

```diff
@@ -0,0 +1,56 @@
+---
+title: Introduction
+order: 1
+---
+
+# Validation
+
+Client side validation is a first-class citizen of ember-headless-form, supporting both [native HTML validation](./native.md) as well as JavaScript-based [custom validation](./custom-validation.md).
+
+See the following chapters for how to integrate the different kinds of validation. Common to all of them is how they affect what happens when a [form is submitted](../usage/data/index.md#getting-data-out). Normally upon submission the form's `@onSubmit` action is called. When however the form has validation in place, it will only be called when the validation actually passed. Otherwise `@onInvalid` is called, which also receives the in this case invalid form data object, alongside an `ErrorRecord` object.
```

Comment:
Another one, feel free to ignore:

```suggestion
See the following chapters for how to integrate the different kinds of validation. Common to all of them is how they affect what happens when a [form is submitted](../usage/data/index.md#getting-data-out). Upon submitting the form, the `@onSubmit` action is called. When the form has validation in place, it will only be called when the validation has passed. Otherwise `@onInvalid` is called, which receives the invalid form data object, alongside an `ErrorRecord` object.
```